### ISSUE TITLE
Simplifying the Import specific Models

### ISSUE THREAD

**Issue by jdalrymple** [jdalrymple/node-gitlab-api#38]
The more I use the library the more i realise the nested structure can be extremly annoying lol

I propose exporting the subclasses directly in the form:

import { Projects }  from 'node-gitlab-api';

const projectsAPI = new Projects(url, token);

projectsAPI.all()
projectsAPI.show(projectId)

etc..

Would this be an improvement?

**Comment by jdalrymple** [comment]
Ill make a branch with this sort of format to get some feedback!

**Comment by arsdehnel** [comment]
I always liked working with https://octokit.github.io/rest.js/ and never felt like it was too annoying to work with the api there.  Pretty much it's _always_ 3 levels deep which is a nice consistency.  I also really like the documentation that's there for it so that might be part of why I like it.

**Comment by jdalrymple** [comment]
Consistency is key, though the one problem with the general rule of going n deep is that some of the resources are quite deep. ie, project repositories commits comments. It would defintely be cleaner to just expose the comments part. hmm maybe like below so you wouldnt have to keep reinitializing the resource you were trying to use.

```javascript
import GitlabAPI from 'node-gitlab-api';

const { Projects, CommitComments } = new GitlabAPI(url, token);

```

**Comment by jdalrymple** [comment]
Working on this right now!

Few major updates:
1. Ability to instatiate the whole api or just the services youre interested in
2. Removed circular references that i somehow didnt see before :joy:
3. Exposing every api as its own standalone service, much cleaner

**Comment by jdalrymple** [comment]
Questions on how to structure certain resources. I've tried my best to put certain resources in their own service ie projecthooks, since there is alot of logic related to them and they are not as required in the main service ie projects. 

but there a few that are relatively small that im unsure about these are:

users -> user keys , user gpg keys , user tokens and user emails
commits -> commit comments (notes)
merge request -> versions

Any thoughts?

What makes the most sense intutively when using the library

User.keys
User.gpgKeys

vs UserKeys.all() (which is consistent with every other service using the .all() function to get all of the results)

**Comment by jdalrymple** [comment]
Hmm, or i can just stick to the organization outlined by the API docs. In the case of custom attributes though, not sure what to do there.

**Comment by jdalrymple** [comment]
Here is the current breakdown:

![1](https://user-images.githubusercontent.com/3743662/37245778-58df1142-246b-11e8-873e-dbd6a8cd4f93.png)
![2](https://user-images.githubusercontent.com/3743662/37245781-5bba9a8a-246b-11e8-8871-b3f33547cff6.png)

**Comment by jdalrymple** [comment]
Which resources here should be combined? Should UserCustomAttributes be within Users? Should RepositoryFiles be within Repositories? etc 

Thanks!

**Comment by jdalrymple** [comment]
After some discussion, i've found a way to get the both of two worlds. That being:

1. Maintaining the same intituive resource structure, even for sub resources
2. Exposing related resources together when convienent.

This can be done by exposing groups. For example, If you need all the functionalities of a User, one could import the UsersGroup which would include {Users, UserKeys, UserGPGKeys, UserEmails, UserCustomAttributes, etc}

Similarly can be done for projects, groups , etc.

**Comment by jdalrymple** [comment]
Released under 3.0.0-rc.2
